### PR TITLE
[FLINK-31140] - Load additional dependencies in operator classpath

### PR DESCRIPTION
**Pull request by tamirsagi**
## What is the purpose of the change

Allow users to add additional dependencies onto operator classpath

## Brief change log

  - Create a target folder with a dedicated environment variable for user dependencies in root dockerFile

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): (yes / no) no
  - The public API, i.e., is any changes to the `CustomResourceDescriptors`: (yes / no)  no
  - Core observer or reconciler logic that is regularly executed: (yes / no)  no

## Documentation

  - Does this pull request introduce a new feature? (yes / no) yes
  - If yes, how is the feature documented? (not applicable / docs / JavaDocs / not documented) docs

### CODE REVIEW

**Comment by tamirsagi** [comment]
@gyfora  Would you please  review?

File: Dockerfile [hunk 1]
```diff
@@ -36,13 +36,16 @@ RUN cd /app/tools/license; mkdir jars; cd jars; \
...
-ENV FLINK_PLUGINS_DIR=/opt/flink/plugins
+ENV FLINK_PLUGINS_DIR=$FLINK_HOME/plugins
 ENV OPERATOR_VERSION=1.5-SNAPSHOT
 ENV OPERATOR_JAR=flink-kubernetes-operator-$OPERATOR_VERSION-shaded.jar
 ENV AUTOSCALER_JAR=flink-kubernetes-operator-autoscaler-$OPERATOR_VERSION.jar
 ENV WEBHOOK_JAR=flink-kubernetes-webhook-$OPERATOR_VERSION-shaded.jar
 ENV FLINK_KUBERNETES_SHADED_JAR=flink-kubernetes-standalone-$OPERATOR_VERSION-shaded.jar
 
+ENV USER_DEPENDENCIES_DIR=$FLINK_HOME/dependencies
+RUN mkdir -p $USER_DEPENDENCIES_DIR
```
Comment (asardaes) [inline Dockerfile:47]: Why not use the same convention as Flink's application mode and name the directory `usrlib`?

Comment (tamirsagi) [reply Dockerfile:47]: I did not want to mix up user jar located under 'userlib' in Flink application and user dependencies on Flink operator.

Comment (asardaes) [reply Dockerfile:47]: Mm I don't have a strong opinion either way, if you think `usrlib` should have different semantics, I think that's fine.

Comment (tamirsagi) [reply Dockerfile:47]: @gyfora  what do you think about this?

Comment (gyfora) [reply Dockerfile:47]: I think this is not usrlib, the operator is controlled by the platform not the user. We could even call it `$FLINK_HOME/operator-lib`

Comment (tamirsagi) [reply Dockerfile]: agree. changed it.

**Pull request merged**